### ISSUE TITLE
Automatic Embargo Length

### ISSUE THREAD

**Issue by wfyson** [eprintsug/hefce_oa#23]
I've had the following question from a repository administrator concerning automatically setting the embargo length:

"if the publication date is set and the embargo period has also been set,  can the system not calculate the embargo length and pull this through to the CC plugin?"

I assume this would be possible to implement although I just thought I should check there isn't a reason why it hasn't been done?

Thanks!

**Comment by jesusbagpuss** [comment]
I can see the logic behind this - but I can also see that the system managing this itself may be a bit scary to some.

Maybe a config flag for 'automatically set embargo when publication date and emb length are known' and a related sub in the config would be good. I'd think that repo admimns might want it to generate some form of email alert - 'I've just set the embargo on x to y' so they can gate-keep it would be sensible.

Things to ponder:
- which document(s) should get an embargo set
- which security flags should get set/altered (e.g. we've implemented a 'hidden' security option - these shouldn't get touched)
- do we take online_publication OR publication dates into consideration
-  ..?

We could implement this in a 'compliance warning' kind of way - the checker could state that it's got enough info to calculate the embargo end date, but it hasn't been set?

**Comment by jesusbagpuss** [comment]
Hi Will.
I think I totally misread this first time around! :hankey: 
I was under the (wrong) impression that we were talking about setting an embargo release date based on the data in the 'hoa_emb_len' field (I still think this would be scary).

What you were _actually_ talking about here is a very sensible thing to do - and I think we should do it!

**Comment by wfyson** [comment]
Ah yes, sorry, I think that confusion was probably caused by the quote I pulled through. The aim here is very much to to set the embargo length field that occurs in the REF CC stage automatically.

Also, off topic, is it possible for me to have push access to this Git repository? I made a few changes on a branch and committed them, and then was going to push before doing a pull request, but then realised I wasn't able to. I could fork the repository, but as I had it all ready to roll I wondered if it would just be easier for me to push my branch?
